File: killrvideo-commons/src/main/java/com/killrvideo/dse/utils/PageableQuery.java
package com.killrvideo.dse.utils;

import com.killrvideo.dse.dto.ResultListPage;

import java.util.Optional;
import java.util.concurrent.CompletableFuture;

@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
public interface PageableQuery<ENTITY> {
    CompletableFuture<ResultListPage<ENTITY>> queryNext(
            Optional<Integer> pageSize,
            Optional<String> pageState,
            Object... queryParams);
}

File: killrvideo-commons/src/main/kotlin/com/killrvideo/dse/utils/PageableQuery.kt
package com.killrvideo.dse.utils

import com.killrvideo.dse.dto.ResultListPage
import java.util.concurrent.CompletableFuture

interface PageableQuery<ENTITY> {
    fun queryNext(
        pageSize: Int?,
        pageState: String?,
        vararg queryParams: Any
    ): CompletableFuture<ResultListPage<ENTITY>>
}
